lib/raptor-cache.js

```diff
@@ -53,18 +53,31 @@ var raptorCache = {
     },
 
     flushAll: function(callback) {
-        var flushTasks = [];
+        if (typeof callback === 'function' || typeof Promise === 'undefined') {
```

Comment:
I had to add this second conditional `... || typeof Promise === 'undefined'` because on line 67, if no callback is provided the old functionality worked by falling back to a default function. If someone is using Node < 4 and calls `flushAll` without a callback, it would break with Promise being undefined without this additional check.